### PR TITLE
Add Sqlite to basic-cli

### PR DESCRIPTION
**Pull request by bhansconnect**
This is basically a direct port of what was in basic-webserver. Just converted to purity inference and separated into it's own crate. The example will not work correctly until https://github.com/roc-lang/roc/pull/7427 lands in the roc repo.

I'm hoping that switching to `libsqlite3-sys` will fix our bundling issues. If not, we can switch back to the linked version of `sqlite3-sys`. They both have 100% compatible apis.

### CODE REVIEW

File: examples/sqlite.roc [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    List.forEachTry!? todo \{ id, task } ->
+        Stdout.line! "\tid: $(id), task: $(task)"
+
+    completed = query_todos_by_status!? db_path "completed"
+
+    Stdout.line!? "\nCompleted Tasks:"
+    List.forEachTry!? completed \{ id, task } ->
+        Stdout.line! "\tid: $(id), task: $(task)"
+
+    Ok {}
```
Comment (lukewilliamboswell) [inline examples/sqlite.roc:22]: Richard requested we avoid the `!?` in examples and use `try` exclusively until with have parens and commas.

Comment (bhansconnect) [reply examples/sqlite.roc:22]: Sad, but ok. I prefer `!?` to `try`, but that is just my person preference.

Comment (bhansconnect) [reply examples/sqlite.roc:22]: Will update

File: platform/Sqlite.roc [hunk 2]
```diff
@@ -0,0 +1,494 @@
...
+u8 = int_decoder Num.toU8Checked
+
+f64 : Str -> SqlDecode F64 [FailedToDecodeReal []]UnexpectedTypeErr
+f64 = real_decoder Ok
+
+f32 : Str -> SqlDecode F32 [FailedToDecodeReal []]UnexpectedTypeErr
+f32 = real_decoder (\x -> Num.toF32 x |> Ok)
+
+# TODO: Mising Num.toDec and Num.toDecChecked
+# dec = realSqlDecoder Ok
```
Comment (lukewilliamboswell) [inline platform/Sqlite.roc:318]: Can we use [`Std.toDec : Str -> Result Dec [InvalidNumStr]`](https://www.roc-lang.org/builtins/Str#toDec)

File: platform/Sqlite.roc [hunk 3]
```diff
@@ -0,0 +1,494 @@
...
+nullable_u8 = nullable_int_decoder Num.toU8Checked
+
+nullable_f64 : Str -> SqlDecode (Nullable F64) [FailedToDecodeReal []]UnexpectedTypeErr
+nullable_f64 = nullable_real_decoder Ok
+
+nullable_f32 : Str -> SqlDecode (Nullable F32) [FailedToDecodeReal []]UnexpectedTypeErr
+nullable_f32 = nullable_real_decoder (\x -> Num.toF32 x |> Ok)
+
+# TODO: Mising Num.toDec and Num.toDecChecked
+# nullable_dec = nullable_real_decoder Ok
```
Comment (lukewilliamboswell) [inline platform/Sqlite.roc:386]: Can we use [`Str.toDec : Str -> Result Dec [InvalidNumStr]`](https://www.roc-lang.org/builtins/Str#toDec)

File: platform/Sqlite.roc (hunk 2 above)
Comment (bhansconnect) [reply platform/Sqlite.roc:318]: I guess we could. Float -> Str -> Dec. That said, we probably should just add `Num.toDec`. for floats, it just means multiplying by `10^18` then casting to an `i128`.

Comment (bhansconnect) [reply platform/Sqlite.roc:318]: We already do this in the zig bitcode. Just isn't wired up/exposed.

File: platform/Sqlite.roc [hunk 4]
```diff
@@ -0,0 +1,710 @@
...
+    else if code == 27 then
+        NOTICE
+    else if code == 28 then
+        WARNING
+    else if code == 100 then
+        ROW
+    else if code == 101 then
+        DONE
+    else
+        crash "unsupported Sqlite error code $(Num.toStr code)"
```
Comment (bhansconnect) [inline platform/Sqlite.roc:670]: While I'm thinking about this. We probably shouldn't crash here. We should instead make a generic error that wraps the code. Some sort of `Unknown` error.

**Review by lukewilliamboswell — APPROVED** [review]

**Comment by bhansconnect** [comment]
I think this is good to merge!

**Comment by Anton-4** [comment]
I'd like to builds this on arm64 linux before we merge this

**Comment by Anton-4** [comment]
I caught up on the zulip conversation and the [successful test](https://github.com/roc-lang/roc/actions/runs/12537623092/job/34961921175?pr=7428) :+1:

File: examples/todos.db
Comment (lukewilliamboswell) [inline examples/todos.db:1]: We changed the todos db? Is this updated here by mistake?

**Review by lukewilliamboswell — APPROVED** [review]

File: examples/todos.db
Comment (Anton-4) [reply examples/todos.db:1]: This happens when running the tests

Comment (bhansconnect) [reply examples/todos.db:1]: This PR added the todos db. So I think everything is good.

**Pull request merged**